packages/js/customer-effort-score/src/feedback-modal/feedback-modal.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { createElement, useState } from '@wordpress/element';
+import PropTypes from 'prop-types';
+import { Button, Modal } from '@wordpress/components';
+import { Text } from '@woocommerce/experimental';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Provides a modal requesting customer feedback.
+ *
+ * Answers and comments are sent to a callback function.
+ *
+ * @param {Object}   props                      Component props.
+ * @param {Function} props.onSendFeedback       Function to call when the results are sent.
+ * @param {string}   props.title                Title displayed in the modal.
+ * @param {string}   props.description          Description displayed in the modal.
+ * @param {string}   props.isSendButtonDisabled Boolean to enable/disable the send button.
+ * @param {string}   props.sendButtonLabel      Label for the send button.
+ * @param {string}   props.cancelButtonLabel    Label for the cancel button.
+ * @param {Function} props.onCloseModal         Callback for when user closes modal by clicking cancel.
+ * @param {Function} props.children             Children to be rendered.
+ */
+function FeedbackModal( {
+	onSendFeedback,
```

Comment:
Nitpicky, but I think we could keep callback names generic and not tie it to the component name.

```suggestion
	onSubmit,
```